orchagent/intfsorch.cpp

```diff
@@ -1056,7 +1061,7 @@ bool IntfsOrch::addRouterIntfs(sai_object_id_t vrf_id, Port &port)
     {
         SWSS_LOG_ERROR("Failed to create router interface %s, rv:%d",
                 port.m_alias.c_str(), status);
-        throw runtime_error("Failed to create router interface.");
+        return handleSaiCreateStatus(SAI_API_ROUTER_INTERFACE, status);
```

Comment:
>handleSaiCreateStatus [](start = 15, length = 21)

To exactly mimic old behavior, you can derive the handleSaiCreateStatus function in this class, and `throw` inside the function.

Reply:
I think in the old behavior (async mode), this throw should not even be called since syncd should already trigger an swss restart before it reaches this throw. To mimic the old behavior close enough, I think a method could be running this handling function before this throw and keep this throw regardless of the return value of the handling function. I updated the code as such.

Reply:
Added a check for the return value of the handling function and throw exception only when it did not successfully handle the failure.